tests/testthat/test-model-compile.R

```diff
@@ -399,4 +398,25 @@ test_that("check_syntax() works with include_paths", {
 
 })
 
+test_that("check_syntax() works with pedantic=TRUE", {
+  skip_on_cran()
+  model_code <- "
+  transformed data {
+    real a;
+    a <- 3;
+  }
+  "
+  stan_file <- write_stan_file(model_code)
+  mod_dep_warning <- cmdstan_model(stan_file, compile = FALSE)
+  expect_message(
+    mod_dep_warning$compile(),
+    "Warning: deprecated language construct used in",
+    fixed = TRUE
+  )
+  expect_message(
+    mod_dep_warning$check_syntax(),
+    "Warning: deprecated language construct used in",
+    fixed = TRUE
+  )
+})
```

Comment:
This is separate from the rest of the PR. This addressed part of https://github.com/stan-dev/cmdstanr/issues/389 by adding a test deprecation warnings show up.